daemon/containerd/service.go

```diff
@@ -139,6 +144,57 @@ func (i *ImageService) GetLayerFolders(img *image.Image, rwLayer layer.RWLayer)
 }
 
 // GetContainerLayerSize returns the real size & virtual size of the container.
-func (i *ImageService) GetContainerLayerSize(containerID string) (int64, int64) {
-	panic("not implemented")
+func (i *ImageService) GetContainerLayerSize(ctx context.Context, containerID string) (int64, int64, error) {
+	snapshotter := i.client.SnapshotService(i.snapshotter)
+	sizeCache := make(map[digest.Digest]int64)
+	snapshotSizeFn := func(d digest.Digest) (int64, error) {
+		if s, ok := sizeCache[d]; ok {
+			return s, nil
+		}
+		usage, err := snapshotter.Usage(ctx, d.String())
+		if err != nil {
+			return 0, err
+		}
+		sizeCache[d] = usage.Size
+		return usage.Size, nil
+	}
+
+	container := i.containers.Get(containerID)
+	if container == nil {
```

Comment:
Not new in this PR, but I wonder why we decided to not make `Get` return a `notFound` error 🤔 (caller can still decide to ignore it, but it's a bit of an odd interface 🤔)